File: GO/AmudancaContinua_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestAmudancaContinua(t *testing.T) {
	tables := []struct {
		cod  float64
		want []string
	}{
		{1.50, []string{"Bom Dia!!", "06", "06", "00"}},
		{95.5, []string{"Boa Tarde!!", "12", "22", "00"}},
		{187.5, []string{"Boa Noite!!", "18", "30", "00"}},
		{279.5, []string{"De Madrugada!!", "00", "38", "00"}},
	}

	for _, table := range tables {
		got := salutationTime(table.cod)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}

File: PHP/AmudancaContinuaTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("AmudancaContinua.php");

class AmudancaContinuaTest extends TestCase {

    public function testAmudancaContinua() {
        $obj = new AmudancaContinua();
        //Test 1
        $want = ["Bom Dia!!","06","06","00"];
        $got = $obj->salutationTime( 1.50 );
        $this->assertEquals($want, $got);
        //Test 2
        $want = ["Boa Tarde!!","12","22","00"];
        $got = $obj->salutationTime( 95.5 );
        $this->assertEquals($want, $got);
        //Test 3
        $want = ["Boa Noite!!","18","30","00"];
        $got = $obj->salutationTime( 187.5 );
        $this->assertEquals($want, $got);
        //Test 4
        $want = ["De Madrugada!!","00","38","00"];
        $got = $obj->salutationTime( 279.5 );
        $this->assertEquals($want, $got);
    }

}
